internal/mesh/internal/controllers/sidecarproxy/builder/destination_builder_test.go

```diff
@@ -98,14 +99,27 @@ func TestBuildExplicitDestinations(t *testing.T) {
 	}
 
 	for name, c := range cases {
-		proxyTmpl := New(testProxyStateTemplateID(), testIdentityRef(), "foo.consul", "dc1", nil).
-			BuildDestinations(c.destinations).
-			Build()
+		t.Run(name, func(t *testing.T) {
```

Comment:
changes in this file:
- use t.Run()
- convert expected and actual proxyTemplates to protos to sort the routers array because of tests flakes based on ordering.